presto-raptor/src/main/java/com/facebook/presto/raptor/systemtables/ColumnRangesSystemTable.java

```diff
@@ -0,0 +1,188 @@
+            return Optional.of(new SchemaTableName(
+                tableName.getSchemaName(),
+                    tableName.getTableName().substring(0, tableName.getTableName().length() - COLUMN_RANGES_TABLE_SUFFIX.length())));
+        }
+        return Optional.empty();
+    }
+
+    private static boolean isColumnRangesTable(SchemaTableName tableName)
+    {
+        return tableName.getTableName().endsWith(COLUMN_RANGES_TABLE_SUFFIX) && tableName.getTableName().length() > COLUMN_RANGES_TABLE_SUFFIX.length();
+    }
+
+    @Override
+    public Distribution getDistribution()
+    {
+        return SINGLE_COORDINATOR;
+    }
+
+    @Override
+    public ConnectorTableMetadata getTableMetadata()
+    {
+        return tableMetadata;
+    }
+
+    @Override
+    public RecordCursor cursor(ConnectorTransactionHandle transactionHandle, ConnectorSession session, TupleDomain<Integer> constraint)
+    {
+        String metadataSqlQuery = getColumnRangesMetadataSqlQuery(sourceTable, indexedRaptorColumns);
+        List<Type> columnTypes = tableMetadata.getColumns().stream()
+                .map(ColumnMetadata::getType)
+                .collect(toImmutableList());
+
+        ResultSetValues resultSetValues = new ResultSetValues(columnTypes);
+
+        try (Connection connection = dbi.open().getConnection();
+                Statement statement = connection.createStatement();
+                ResultSet resultSet = statement.executeQuery(metadataSqlQuery)) {
+            if (resultSet.next()) {
+                resultSetValues.extractValues(resultSet, ImmutableSet.of(), ImmutableSet.of());
+                List<Object> record = IntStream.range(0, columnTypes.size())
+                        .boxed()
+                        .map(field -> getColumnValue(resultSetValues, field, columnTypes.get(field)))
+                        .collect(toList());
+                return new InMemoryRecordSet(columnTypes, ImmutableList.of(record)).cursor();
+            }
+            else {
+                return new InMemoryRecordSet(columnTypes, ImmutableList.of()).cursor();
+            }
+        }
+        catch (SQLException | DBIException e) {
+            throw metadataError(e);
+        }
+    }
+
+    private Object getColumnValue(ResultSetValues resultSetValues, int field, Type columnType)
+    {
+        Class<?> javaType = columnType.getJavaType();
+        if (resultSetValues.isNull(field)) {
+            return null;
+        }
+
+        if (javaType == double.class) {
+            return resultSetValues.getDouble(field);
+        }
+        else if (javaType == long.class) {
+            return resultSetValues.getLong(field);
+        }
+        else if (javaType == boolean.class) {
+            return resultSetValues.getBoolean(field);
+        }
+        else if (javaType == Slice.class) {
+            return resultSetValues.getSlice(field);
+        }
+        else {
+            throw new VerifyException("Unknown Java type: " + javaType);
```

Comment:
This should do `javaType.getName()` since `toString()` for `Class` has extra stuff